### ISSUE TITLE
Style/FormatStringToken doesn't work inside a hash

### ISSUE THREAD

**Issue by 0leynikov** [rubocop-hq/rubocop#7407]
## Expected behavior

Given a file foobar.rb:

```ruby
# frozen_string_literal: true

format('%{foo}', foo: 'foo')
{ bar: format('%{foo}', foo: 'foo') }
```

I expect to get two offenses:
```
foobar.rb:3:9: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over template tokens (like %{foo}).
format('%{foo}', foo: 'foo')
        ^^^^^^
foobar.rb:4:16: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over template tokens (like %{foo}).
{ bar: format('%{foo}', foo: 'foo') }
               ^^^^^^
```

## Actual behavior

I only get the first one of them. `Style/FormatStringToken` offenses from inside a hash aren't reported.

```
foobar.rb:3:9: C: Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over template tokens (like %{foo}).
format('%{foo}', foo: 'foo')
        ^^^^^^
```

## Steps to reproduce the problem

With a file foobar.rb from above, execute:

```
$ rubocop foobar.rb
```

## RuboCop version

```
$ rubocop -V
0.75.0 (using Parser 2.6.4.1, running on ruby 2.5.3 x86_64-linux)
```

**Comment by 0leynikov** [comment]
The bug also holds with unannotated tokens. Try to replace foobar.rb contents with the following:

```ruby
# frozen_string_literal: true

format('%s', 'foo')
{ bar: format('%s', 'foo') }
```

Again, only the first offense is detected.

**Comment by buehmann** [comment]
Thanks. That behaviour is a side effect of this old change: https://github.com/rubocop-hq/rubocop/pull/5634

I will take a look at this later.
